### PR TITLE
Create new home hero molecule

### PR DESCRIPTION
**Pull request by jimmynotjim**
Updates the home hero to become an abstracted molecule.

## Additions

- Adds home_hero macro that takes a value with the required settings
- Adds home_hero styles including change to how the images are handled

## Changes

- Moves full well modifier to wells.less and documents usage
- Cleans up the homepage specific styles file

## Testing

- `gulp build` and check out the home page. It should look exactly the same.

## Review

- @schaferjh 
- @anselmbradford 
- @sebworks 
- @KimberlyMunoz 

## Screenshots

screen shot 2016-03-22 at 3 51 03 pm

screen shot 2016-03-22 at 3 51 11 pm

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by schaferjh** [comment]
Score. Just to double check (since I'm only going off the screenshots) -- the height and text spacing match the [DM for illustrated heroes] (https://cfpb.github.io/design-manual/ui-toolkit/heroes.html), right?

Note for the BEWD task - content managers sill need to be able to select white or black text depending on the coloring of the background image. _I'm just talking out loud on that one, I'll remember to add it to the BEWD task._

File: cfgov/unprocessed/css/molecules/overlay-hero.less [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  tags:
+    - cfgov-molecules
+*/
+
+.m-overlay-hero {
+    position: relative;
+
+    padding-top: unit( @grid_gutter-width / @base-font-size-px, em );
+    padding-right: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
+    padding-left: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/overlay-hero.less]: Is the bottom padding set? Could this be:

```
padding: unit( @grid_gutter-width / @base-font-size-px, em );
padding-bottom: 0;
```

File: cfgov/unprocessed/css/molecules/overlay-hero.less [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        color: @white;
+    } );
+
+    .respond-to-min( @bp-xl-min, {
+        padding-right: unit( 45px / @base-font-size-px, em );
+        padding-left: unit( 45px / @base-font-size-px, em );
+
+        // Max-width - both gutters: required due to well spanning the
+        // full page
+        max-width: 1230px - ( 45px * 2 );
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/overlay-hero.less]: Maybe pull 45px out as a variable. Looks like it's `@grid_gutter-width + ( @grid_gutter-width / 2 )`

File: cfgov/unprocessed/css/molecules/overlay-hero.less [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+        .respond-to-min( @bp-sm-min, {
+            .heading-2();
+        } );
+    }
+
+    &_sm-img {
+        .u-flexible-container();
+
+        background-position: center center;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/overlay-hero.less]: Minor: Could just be one `center` I think, like how repeat is one `no-repeat`.

File: cfgov/unprocessed/css/organisms/well.less [hunk 4]
```diff
@@ -48,6 +58,18 @@
         padding-left: unit( @grid_gutter-width / @base-font-size-px, em );
         padding-right: unit( @grid_gutter-width / @base-font-size-px, em );
     } );
+
+    &__full {
+        .respond-to-min( @bp-xl-min, {
+            // TODO: 45px should be moved to a global grid gutter var
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/well.less]: Ahh

**Comment by KimberlyMunoz** [comment]
:+1: pending addressing the feedback above.

File: cfgov/unprocessed/css/molecules/overlay-hero.less (hunk 1 above)
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/overlay-hero.less]: No, it's never set, why set it from default just to reset default?

**Comment by jimmynotjim** [comment]
@schaferjh 

> Score. Just to double check (since I'm only going off the screenshots) -- the height and text spacing match the DM for illustrated heroes, right?

It matches the original specs for the homepage. Should the homepage change to match the Illustration specs or do we need two versions of this? I'd suggest one version so that we can keep things simple and organized.

> Note for the BEWD task - content managers sill need to be able to select white or black text depending on the coloring of the background image. I'm just talking out loud on that one, I'll remember to add it to the BEWD task.

I'll need to add that option here as well. I'll do that with other cleanup in the next push.

File: cfgov/unprocessed/css/molecules/overlay-hero.less (hunk 1 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/overlay-hero.less]: Nevermind. I thought all three were set to the same value, but seeing now that right and left are half gutters.

**Comment by schaferjh** [comment]
Actually, we don't want it to match the specs for the original home page hero --  rather, we're trying to add an image to the standard hero. So the home page hero stays unique, and the standard hero has another version -- one with an image in the background. Apologies if that wasn't clear earlier.

**Comment by jimmynotjim** [comment]
Sorry for the confusion. Rather than try to fit the home hero into the existing hero, I've updated this PR to be more of an abstraction of the home-hero from the page styles. I think it's still valuable to make it atomic, even if it's not being reused. 

@KimberlyMunoz @anselmbradford @sebworks 
Ready for review again

**Comment by KimberlyMunoz** [comment]
:+1:

**Comment by sebworks** [comment]
:+1:

**Pull request merged**